--- a/lib/luks2/luks2.h
+++ b/lib/luks2/luks2.h
@@ -83,7 +83,7 @@
 /*
  * LUKS2 header in-memory.
  */
-typedef struct json_object json_object;
+struct json_object;
 struct luks2_hdr {
 	size_t		hdr_size;
 	uint64_t	seqid;
@@ -94,7 +94,7 @@
 	uint8_t		salt1[LUKS2_SALT_L];
 	uint8_t		salt2[LUKS2_SALT_L];
 	char		uuid[LUKS2_UUID_L];
-	json_object	*jobj;
+	struct json_object	*jobj;
 };
 
 /*
@@ -135,8 +135,8 @@
 		      struct luks2_hdr *hdr,
 		      const char *backup_file);
 
-uint64_t LUKS2_hdr_and_areas_size(json_object *jobj);
-uint64_t LUKS2_keyslots_size(json_object *jobj);
+uint64_t LUKS2_hdr_and_areas_size(struct json_object *jobj);
+uint64_t LUKS2_keyslots_size(struct json_object *jobj);
 
 /*
  * Generic LUKS2 keyslot
