--- a/src/cryptsetup.c
+++ b/src/cryptsetup.c
@@ -23,6 +23,7 @@
 
 #include "cryptsetup.h"
 #include <uuid/uuid.h>
+#include "posix_fallocate.h"
 
 static const char *opt_cipher = NULL;
 static const char *opt_hash = NULL;
--- a/lib/utils_device.c
+++ b/lib/utils_device.c
@@ -39,6 +39,7 @@
 #endif
 #include "internal.h"
 #include "utils_device_locking.h"
+#include "posix_fallocate.h"
 
 struct device {
 	char *path;
--- a/lib/posix_fallocate.h
+++ b/lib/posix_fallocate.h
@@ -0,0 +1,27 @@
+#ifndef	_POSIX_FALLOCATE_H
+#define	_POSIX_FALLOCATE_H	1
+
+#include <stdint.h>
+#include <sys/syscall.h>
+
+/********************************************************************************/
+/* stole from uClibc-33a12b5540b8abbc4ee0ecb3a51912b3c7868517: include/endian.h */
+#define OFF_HI(offset) (offset >> 31)
+#define OFF_LO(offset) (offset)
+#define OFF64_HI(offset) (uint32_t)(offset >> 32)
+#define OFF64_LO(offset) (uint32_t)(offset & 0xffffffff)
+#define OFF_HI_LO(offset) __LONG_LONG_PAIR(OFF_HI(offset), OFF_LO(offset))
+#define OFF64_HI_LO(offset) __LONG_LONG_PAIR(OFF64_HI(offset), OFF64_LO(offset))
+/********************************************************************************/
+
+inline int posix_fallocate64(int __fd, __off64_t __offset, __off64_t __len)
+{
+  return syscall(SYS_fallocate, __fd, 0, OFF64_HI_LO (__offset), OFF64_HI_LO (__len));
+}
+
+inline int posix_fallocate(int __fd, __off64_t __offset, __off64_t __len)
+{
+  return syscall(SYS_fallocate, __fd, 0, OFF64_HI_LO (__offset), OFF64_HI_LO (__len));
+}
+
+#endif /* posix_fallocate.h  */
